deploy/operator_template.yaml

```diff
@@ -14,17 +15,14 @@ spec:
         name: pulumi-kubernetes-operator
     spec:
       serviceAccountName: pulumi-kubernetes-operator
-      imagePullSecrets:
```

Comment:
I seem to have not explained why some of these changes were ok.

The image pull secret was a remnant of the internal preview where the image was published to a private dockerhub repo. This is no longer needed since we publish a public image.